### PR TITLE
Add support for verifying images

### PR DESCRIPTION
**Pull request by fishnix**
Might make sense to make this a GET endpoint if we want to return meaningful error, but this is simple and works for now.  This could change substantially if we are passing some form of authentication or auth pointer.

### CODE REVIEW

**Comment by damnski** [comment]
Nice breadth of testing scenarios ;-)  Code testing failing in registry/ - i.e. registry/verifier.go:86:14: undefined: http.NewRequestWithContext

**Comment by fishnix** [comment]
your golang version is too old @damnski

File: registry/verifier.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	if header == "" {
+		return "", errors.New("empty authenticate header")
+	}
+	log.Debugf("parsing bearer token header: %s", header)
+
+	realm := ""
+	scope := ""
+	service := ""
+
+	value := strings.TrimPrefix(header, "Bearer ")
```
[the remark this answers is not in the document]
Comment (fishnix) [reply registry/verifier.go]: yeah this function could probably use some cleanup.... this was the "make it work" stage 😆 I'll look at it.

File: registry/verifier.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	if code > 299 && code <= 399 {
+		log.Warnf("got unexpexted redirection code for %s during verification %d", v.Name, code)
+	}
+
+	if code > 499 {
+		msg := fmt.Sprintf("verify failed, status: %d", code)
+		return false, errors.New(msg)
+	}
+
+	if code > 299 && code <= 499 {
```
[the remark this answers is not in the document]
Comment (fishnix) [reply registry/verifier.go]: I don't want to return true if I got a redirect though, right?  We aren't currently following redirects.

[the remark this answers is not in the document]
Comment (tenyo) [reply registry/verifier.go]: ok, I see

File: registry/verifier_test.go [hunk 3]
```diff
@@ -322,4 +322,32 @@ func TestBearerTokenAuth(t *testing.T) {
...
+	_, err = testVerifier.bearerTokenAuth(context.TODO(), "\"realm=foobar\",\"scope=foo\",\"service=bar\"")
+	if err == nil {
+		t.Error("expected error, got nil")
+	}
+	t.Logf("got err response for missing Bearer '%s'", err)
+
+	// bad json response
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("{fuuuuuuuu"))
```
Comment (tenyo) [inline registry/verifier_test.go]: 😮

**Comment by tenyo** [comment]
Looks good 👍

**Pull request merged**